backend/src/database/repositories/memberRepository.ts

```diff
@@ -2313,7 +2170,237 @@ class MemberRepository {
+    ['score', 'm.score'],
+    ['averageSentiment', 'msa."averageSentiment"'],
+    ['activityTypes', 'msa."activityTypes"'],
+    ['activeOn', 'msa."activeOn"'],
+    ['activityCount', 'msa."activityCount"'],
+
+    // others
+    ['organizations', 'mo."organizationId"'],
+
+    // fields for querying
+    ['attributes', 'm.attributes'],
+  ])
+
   static async findAndCountAll(
+    {
+      filter = {} as any,
+      limit = 20,
+      offset = 0,
+      orderBy = 'joinedAt_DESC',
+      segmentId = undefined,
+      countOnly = false,
+      fields = [...MemberRepository.QUERY_FILTER_COLUMN_MAP.keys()],
+      include = {
+        identities: true,
+        segments: false,
+        lfxMemberships: false,
+        memberOrganizations: false,
+      } as {
+        identities?: boolean
+        segments?: boolean
+        lfxMemberships?: boolean
+        memberOrganizations?: boolean
+      },
+      attributesSettings = [] as AttributeData[],
+    },
+    options: IRepositoryOptions,
+  ) {
+    const transaction = SequelizeRepository.getTransaction(options)
+
+    if (!attributesSettings) {
+      attributesSettings = (await MemberAttributeSettingsRepository.findAndCountAll({}, options))
+        .rows
+    }
+
+    const qx = SequelizeRepository.getQueryExecutor(options, transaction)
+
+    const withAggregates = !!segmentId
+    let segment
+    if (withAggregates) {
+      segment = await new SegmentRepository(options).findById(segmentId)
+
+      if (segment === null) {
+        options.log.info('No segment found for member query. Returning empty result.')
+        return {
+          rows: [],
+          count: 0,
+          limit,
+          offset,
+        }
+      }
+    }
+
+    const params = {
+      limit,
+      offset,
+      tenantId: options.currentTenant.id,
+      segmentId: segment?.id,
+    }
+
+    const filterString = RawQueryParser.parseFilters(
+      filter,
+      MemberRepository.QUERY_FILTER_COLUMN_MAP,
+      [],
+      params,
+      { pgPromiseFormat: true },
+    )
+
+    const order = (function prepareOrderBy(
+      orderBy = withAggregates ? 'activityCount_DESC' : 'id_DESC',
+    ) {
+      const orderSplit = orderBy.split('_')
+
+      const orderField = MemberRepository.QUERY_FILTER_COLUMN_MAP.get(orderSplit[0])
+      if (!orderField) {
+        return withAggregates ? 'msa."activityCount" DESC' : 'm.id DESC'
+      }
+      const orderDirection = ['DESC', 'ASC'].includes(orderSplit[1]) ? orderSplit[1] : 'DESC'
+
+      return `${orderField} ${orderDirection}`
+    })(orderBy)
+
+    const createQuery = (fields) => `
+      SELECT
+        ${fields}
+      FROM members m
+      ${
+        withAggregates
+          ? `LEFT JOIN "memberSegmentsAgg" msa ON msa."memberId" = m.id AND msa."segmentId" = $(segmentId)`
+          : ''
+      }
+      ${
+        include.memberOrganizations
+          ? `LEFT JOIN "memberOrganizations" mo ON mo."memberId" = m.id`
+          : ''
+      }
+      WHERE 1=1
```

Comment:
do we need this? there are already some default filters

Reply:
True, we don't